src/Professions/Professions.java

```diff
@@ -0,0 +1,62 @@
+package Professions;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.function.BiFunction;
+
+/**
+ * Keeps track of all static data and helper methods.
+ * Responsible for static factory initialization of players, resolved by profession.
+ * @author Aya
+ *
+ */
+public class Professions {
+	public static final Set<String> ALLOWED_PROFESSION_NAMES;
+	private static final List<String> ALLOWED_PROFESSION_NAMES_INDEXABLE;

```

Comment:
List for efficient indexing for random profession generation. `O(1)` for indexing on a list, whereas indexing into a set is `O(n)`.